lib/API/Startup.js

```diff
@@ -89,51 +90,51 @@ module.exports = function(CLI) {
       return cb();
     }
 
-    if (fs.existsSync('/etc/init.d/pm2-init.sh')) {
+    if (fs.existsSync('/etc/init.d/' + service_name + '-init.sh')) {
       platform = 'oldsystem';
     }
 
     switch(platform) {
     case 'systemd':
       commands = [
-        'systemctl stop pm2',
-        'systemctl disable pm2',
-        'systemctl status pm2',
-        'rm /etc/systemd/system/pm2.service'
+        'systemctl stop ' + service_name,
+        'systemctl disable ' + service_name,
+        'systemctl status ' + service_name,
+        'rm /etc/systemd/system/' + service_name + '.service'
       ];
       break;
     case 'systemv':
       commands = [
-        'chkconfig pm2 off',
-        'rm /etc/init.d/pm2'
+        'chkconfig ' + service_name + ' off',
+        'rm /etc/init.d/' + service_name
       ];
       break;
     case 'oldsystem':
       Common.printOut(cst.PREFIX_MSG + 'Disabling and deleting old startup system');
       commands = [
-        'update-rc.d pm2-init.sh disable',
-        'update-rc.d -f pm2-init.sh remove',
-        'rm /etc/init.d/pm2-init.sh'
+        'update-rc.d ' + service_name + '-init.sh disable',
+        'update-rc.d -f ' + service_name + '-init.sh remove',
+        'rm /etc/init.d/' + service_name + '-init.sh'
       ];
       break;
     case 'openrc':
       commands = [
-	      '/etc/init.d/pm2 stop',
-        'rc-update delete pm2 default',
-        'rm /etc/init.d/pm2'
+	      '/etc/init.d/' + service_name + ' stop',
+        'rc-update delete ' + service_name + ' default',
+        'rm /etc/init.d/' + service_name
       ];
       break;
     case 'upstart':
       commands = [
-        'update-rc.d pm2 disable',
-        'update-rc.d -f pm2 remove',
-        'rm /etc/init.d/pm2'
+        'update-rc.d ' + service_name + ' disable',
+        'update-rc.d -f ' + service_name + ' remove',
+        'rm /etc/init.d/' + service_name
       ];
       break;
     case 'launchd':
-      var destination = path.join(process.env.HOME, 'Library/LaunchAgents/PM2.plist');
+      var destination = path.join(process.env.HOME, 'Library/LaunchAgents/' + service_name.toUpperCase() + '.plist');
```

Comment:
On osx lowercase is fine but it should be something like this: `PM2.soyuka.plist` i.e dot-separated instead of dash.